src/Controls/src/Core/Compatibility/Handlers/NavigationPage/iOS/NavigationRenderer.cs

```diff
@@ -1719,7 +1719,11 @@ public override async void DidMoveToParentViewController(UIViewController parent
 				{
 					NavigationRenderer navRenderer;
 					if (_navigation.TryGetTarget(out navRenderer))
+					{
+						if (!navRenderer._uiRequestedPop)
```

Comment:
if I comment out this code the test still passes

Reply:
@PureWeen, I modified the test case. Could you please review it ?